DDD-7.md

```diff
@@ -0,0 +1,153 @@
+While in the first case the connector life cycle and execution of connector tasks are managed by Kafka Connect, in the later case the task life cycle is managed purely by the Debezium engine itself and the Debezium project has full control how the tasks within the engine are executed.
+Current implementation of the [`DebeziumEngine` interface](https://github.com/debezium/debezium/blob/v2.4.0.Final/debezium-api/src/main/java/io/debezium/engine/DebeziumEngine.java), [`EmebeddedEngine`](https://github.com/debezium/debezium/blob/v2.4.0.Final/debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java#L91), executes all the steps in a serial manner.
+This includes executing event transformations as well as event serialization.
+Event serialization is not provided directly by `EmebddedEngine`.
+It's provided by its extension, [`ConvertingEngine`](https://github.com/debezium/debezium/blob/v2.4.0.Final/debezium-embedded/src/main/java/io/debezium/embedded/ConvertingEngineBuilder.java#L169), which implements it as part of [event processing](https://github.com/debezium/debezium/blob/v2.4.0.Final/debezium-embedded/src/main/java/io/debezium/embedded/ConvertingEngineBuilder.java#L102).
+
+Moreover, current `EmebddedEngine` implementation doesn't support executing multiple source tasks, even in case the source connector supports it, such e.g. [SQL server connector](https://debezium.io/documentation/reference/2.4/connectors/sqlserver.html#sqlserver-property-tasks-max).
+Only the [first task is executed](https://github.com/debezium/debezium/blob/v2.4.0.Final/debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java#L918), no matter how many tasks connector configuration provides.
+
+In the era of large data sets and multicore servers, using a single thread for processing all the CDC events from the database is a clear limiting factor.
+Providing a new implementation of `DebeziumEngine` interface which would run some of the tasks in parallel can give significant performance boost.
+New implementation and related changes should also aim for a good test coverage and ease of testing of the new implementation as well as any other future implementations.
+
+## Goals
+
+Provide new implementation of [`DebeziumEngine` interface](https://github.com/debezium/debezium/blob/v2.4.0.Final/debezium-api/src/main/java/io/debezium/engine/DebeziumEngine.java) which would
+
+* allow to run multiple [source tasks](https://github.com/apache/kafka/blob/3.6.0/connect/api/src/main/java/org/apache/kafka/connect/source/SourceTask.java) for given connector if the connector provides multiple tasks
+* run potentially time-consuming code (e.g. event transformation or serialization) in the dedicated threads
+* allow possible further speedup by optionally disabling total ordering of the messages
```

Comment:
These are interesting ideas that could potentially be implemented in Kafka Connect too. I created https://issues.apache.org/jira/browse/KAFKA-15912 to gauge the feasibility.